src/Unused/Types.hs

```diff
@@ -2,17 +2,49 @@ module Unused.Types
     ( TermMatch(..)
     , TermMatchSet
     , ParseResponse(..)
+    , responseFromParse
+    , withOneFile
+    , withOneOccurrence
+    , skipRailsExtras
     ) where
 
 import Text.Parsec (ParseError)
-import Data.Map.Strict as Map
+import qualified Data.Map.Strict as Map
+import Data.List (isInfixOf)
+import Unused.Util (groupBy)
 
 data TermMatch = TermMatch
     { term :: String
     , path :: String
     , occurrences :: Int
     } deriving Show
 
-type TermMatchSet = Map String [TermMatch]
+type TermMatchSet = Map.Map String [TermMatch]
+
 
 data ParseResponse = ValidParse TermMatchSet | InvalidParse ParseError
+
+responseFromParse :: Either ParseError [TermMatch] -> ParseResponse
+responseFromParse =
+    either InvalidParse (ValidParse . Map.fromList . groupBy term)
+
+withOneFile :: ParseResponse -> ParseResponse
+withOneFile = applyFilterToMap (\_ a -> length a == 1)
+
+withOneOccurrence :: ParseResponse -> ParseResponse
+withOneOccurrence = applyFilterToMap (\_ a -> (sum $ fmap occurrences a) == 1)
+
+skipRailsExtras :: ParseResponse -> ParseResponse
+skipRailsExtras = notMatchingPath "db/migrate" . notMatchingPath "app/controllers"
+
+notMatchingPath :: String -> ParseResponse -> ParseResponse
+notMatchingPath s =
+    applyFilterToMatches $ not . isInfixOf s . path
+
+applyFilterToMap :: (String -> [TermMatch] -> Bool) -> ParseResponse -> ParseResponse
+applyFilterToMap _ (InvalidParse e) = InvalidParse e
+applyFilterToMap f (ValidParse s) = ValidParse $ Map.filterWithKey f s
+
+applyFilterToMatches :: (TermMatch -> Bool) -> ParseResponse -> ParseResponse
+applyFilterToMatches _ (InvalidParse e) = InvalidParse e
+applyFilterToMatches f (ValidParse s) = ValidParse $ Map.map (filter f) s
```

Comment:
If you made `ParseResponse` a `Functor`, you could use `fmap` anywhere you're saying "do a thing to a valid parse, and pass along any invalid one".